src/ntfsck.c

```diff
@@ -3536,7 +3536,7 @@ static int ntfsck_scan_index_entries_btree(ntfs_volume *vol)
 			ictx = NULL;
 		}
 
-		if (dir_ni->fsck_ibm) {
+		if (dir_ni && dir_ni->fsck_ibm) {
```

Comment:
Is it real issue ? dir_ni was already checked before queuing ntfs_dir_list.

Reply:
After intfs_inode_open() fails and dir_ni is NULL in the loop, the code flow goes to this condition. This happens when a disk is unplugged.

Reply:
can you update the patch description ?

Reply:
I have done.